modules/Bio/EnsEMBL/Hive/DBSQL/AnalysisJobAdaptor.pm

```diff
@@ -587,6 +587,59 @@ sub reset_or_grab_job_by_dbID {
     return $job;
 }
 
+=head2 reset_job_by_input_id_and_sync
+
+  Arg [1]: string $input_id pattern
+  Arg [2]: array $analsyes_object_array
+  Example:
+    my $job = $self->{'dba'}->get_AnalysisJobAdaptor->reset_job_by_input_id_and_sync($input_id_pattern, $analsyes_object_array);
+  Description:
+    Reset jobs for the specified $input_id pattern and $analsyis_object_array. $input_id can be a wildcard argument.
+  Returntype : none
+  Exceptions :
+  Caller     : beekeeper.pl
+
+=cut
+
+sub reset_job_by_input_id_and_sync {
+    my ($self, $input_id_pattern, @analyses_list) = @_;
+
+    #Convert MySQL wildcard pattern to Perl wildcard pattern
+    if ($input_id_pattern =~ /\A[^%]/) {
+        $input_id_pattern = '^'.$input_id_pattern;
+    }
+    if ($input_id_pattern =~ /[^%]\z/) {
+        $input_id_pattern = $input_id_pattern.'$';
+    }
```

Comment:
See https://github.com/Ensembl/ensembl-hive/blob/version/2.5/modules/Bio/EnsEMBL/Hive/Utils/Collection.pm#L176-L179 which implements "analyses pattern" with `%` too.
 `^` and `$` are added regardless of whether the requested pattern starts or ends with `%`. I don't know if it makes a difference in terms of performance: your solution is certainly more accurate but we could just not bother and _always_ add `^` and `$` in the `qr` expression, and save 6 lines of code.
Or should we change Collection.pm ?
Or factor this out in Utils.pm ? Add there a simple function that takes a user pattern and returns the perl `qr` object, which could be used in both this file and Collection.pm . Actually, I like this idea ! It would make it easier to extend the language if we ever want to !